### PR TITLE
Update AzDO API calls to use unified version

### PR DESCRIPTION
**Pull request by akoeplinger**
We were using a 5.x-preview version in a few places, update and unify to the released 6.0 where possible.

### CODE REVIEW

**Comment by riarenas** [comment]
Azdo's handling of API versions is not quite as advertised. We did some work in this area in https://github.com/dotnet/arcade/pull/5316, and found out that some APIs simply never make it out of preview.

File: scripts/duplicate-feed.ps1 [hunk 1]
```diff
@@ -46,7 +46,7 @@ function Get-Package-List($vstsAuthHeader, $account, $visibility, $feed) {
     # Get the list of packages from the source and target
     $listOfPackages = @()
     try {
-        $packageListUri = "https://feeds.dev.azure.com/$account/${visibility}_apis/packaging/Feeds/$feed/packages?api-version=5.1-preview.1"
+        $packageListUri = "https://feeds.dev.azure.com/$account/${visibility}_apis/packaging/Feeds/$feed/packages?api-version=6.0"
```
Comment (riarenas) [inline scripts/duplicate-feed.ps1:49]: Calling this api, for example https://feeds.dev.azure.com/dnceng/public/_apis/packaging/Feeds/dotnet-5/packages?api-version=6.0 returns an error:

```
The requested version \"6.0\" of the resource is under preview. The -preview flag must be supplied in the api-version for such requests. For example: \"6.0-preview\
```

File: src/Microsoft.DotNet.Helix/Sdk/CheckAzurePipelinesTestResults.cs [hunk 2]
```diff
@@ -128,7 +128,7 @@ private async Task CheckTestResultsWithFlakySupport(HttpClient client)
                 {
                     using (var req = new HttpRequestMessage(
                         HttpMethod.Get,
-                        $"{CollectionUri}{TeamProject}/_apis/test/resultsummarybybuild?buildId={BuildId}&api-version=5.1-preview.2")
+                        $"{CollectionUri}{TeamProject}/_apis/test/resultsummarybybuild?buildId={BuildId}&api-version=6.0")
```
Comment (riarenas) [inline src/Microsoft.DotNet.Helix/Sdk/CheckAzurePipelinesTestResults.cs:131]: This call also fails unless preview version is used:

https://dev.azure.com/dnceng/public/_apis/test/resultsummarybybuild?buildId=926602&api-version=6.0

**Comment by riarenas** [comment]
We need to be very careful about these updates, we previously got in trouble in this repo and in arcade-services by thinking that when a version is marked as released, it means everything that was previously in preview made it to the release version. The GET apis are easy to test, but we'll need extra validation for the POST and PATCH operations.

**Comment by markwilkie** [comment]
Is there a big reason to get this in before the new year?  I'm a little hesitant to make this kind of potentially breaking change without a more full crew.

**Comment by akoeplinger** [comment]
Ah, I didn't realize AzDO is being unconventional with their preview APIs. There's no reason to take this right now, I just saw an API call in a build that failed with `503 Service Unavailable` and noticed the preview API being used.

It looks like for these preview APIs it doesn't matter which version you use as long as there's `-preview` in the string, I'll put that back and we can decide when/if to take it in the new year.

**Comment by markwilkie** [comment]
is there a reason you're suggesting to go to 6 preview instead of a stable version of 5?

**Comment by akoeplinger** [comment]
@markwilkie I updated to `6.0-preview` everywhere where we used some `-preview` API because as discussed above AzDO gates some of the APIs behind `-preview` but doesn't really care whether it's `5.0-preview` or `6.0-preview`. All of the stable APIs where updated to `6.0`

**Comment by markwilkie** [comment]
Makes sense.  I think we should probably build this version against the big repos to make sure we're not seeing something unexpected.  https://github.com/dotnet/arcade/blob/main/Documentation/Policy/TestingMSBuildGuidance.md

**Comment by markwilkie** [comment]
Sorry....better link:  https://github.com/dotnet/arcade/blob/main/Documentation/Validation/Overview.md

**Comment by ChadNedzlek** [comment]
This has conflicts, so I'm going to close this out. We can reopen and revisit if this is high priority.

**Pull request closed without merging**